jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/AbstractPathSpec.java

```diff
@@ -55,7 +60,7 @@ public final boolean equals(Object obj)
     @Override
     public final int hashCode()
     {
-        return Objects.hash(getDeclaration());
+        return Objects.hash(this.getClass().getSimpleName() + ":" + getDeclaration());
```

Comment:
Implementation of `hashCode()` should be based on the same items used for `equals()`.
Consider:

```suggestion
        return Objects.hash(getGroup().ordinal(), getSpecLength(), getDeclaration(), getClass().getName());
```